processing/src/main/java/org/apache/druid/segment/VirtualColumns.java

```diff
@@ -59,7 +59,7 @@
 /**
  * Class allowing lookup and usage of virtual columns.
  */
-public final class VirtualColumns implements Cacheable
+public class VirtualColumns implements Cacheable
```

Comment:
qq: Why are we reverting this change?

Reply:
oh, I didn't really mean to push it in the first place. I thought about it briefly, but then was thinking that it might be nice to someday split out `VirtualColumns` with "dot" support from `VirtualColumns` without it, so that we don't have to check every `getVirtualColumn` call to see if it contains a `.` to split on and check the "dot" virtual columns. These are not used by built-in virtual column implementations, only some extensions, so I think we could specialize implementations a bit to only do this stuff if we detect any `VirtualColumn.usesDotNotation` that return true.